src/main/java/net/dries007/tfc/client/render/blockentity/MoldBlockEntityRenderer.java

```diff
@@ -0,0 +1,194 @@
+import net.dries007.tfc.common.blockentities.MoldBlockEntity;
+import net.dries007.tfc.common.component.TFCComponents;
+import net.dries007.tfc.common.component.fluid.FluidComponent;
+import net.dries007.tfc.common.component.mold.IMold;
+import net.dries007.tfc.common.items.TFCItems;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.blaze3d.vertex.VertexConsumer;
+
+import net.dries007.tfc.TerraFirmaCraft;
+import net.dries007.tfc.client.RenderHelpers;
+import net.dries007.tfc.common.recipes.CastingRecipe;
+import net.dries007.tfc.common.recipes.HeatingRecipe;
+import net.dries007.tfc.util.MetalItem;
+import net.dries007.tfc.util.collections.IndirectHashCollection;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.renderer.blockentity.BlockEntityRenderer;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.client.resources.model.BakedModel;
+import net.minecraft.client.resources.model.ModelResourceLocation;
+import net.minecraft.core.Direction;
+import net.minecraft.core.registries.BuiltInRegistries;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.crafting.RecipeManager;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.phys.AABB;
+import net.minecraft.world.phys.Vec3;
+import net.neoforged.neoforge.client.extensions.common.IClientFluidTypeExtensions;
+import net.neoforged.neoforge.fluids.FluidStack;
+
+public class MoldBlockEntityRenderer implements BlockEntityRenderer<MoldBlockEntity> {
+
+    @Override
+    public void render(MoldBlockEntity mold, float partialTicks, PoseStack poseStack, MultiBufferSource buffer,
+            int combinedLight, int combinedOverlay) {
+        VertexConsumer builder = buffer.getBuffer(RenderType.cutout());
+
+        // Render flow into the mold
+        if (mold.hasSource()) {
+            ResourceLocation texture = IClientFluidTypeExtensions.of(mold.getSourceFluid().getFluidType())
+                    .getStillTexture();
+            int color = RenderHelpers.getFluidColor(mold.getSourceFluid());
+            TextureAtlasSprite sprite = Minecraft.getInstance().getTextureAtlas(RenderHelpers.BLOCKS_ATLAS)
+                    .apply(texture);
+
+            RenderHelpers.renderChannelFlow(poseStack, builder, sprite, color, combinedLight, combinedOverlay,
+                    mold.getFlowSource(), true);
+            if (mold.getFlowSource().getLeft() == Direction.UP) {
+                RenderHelpers.renderChannelFlowCenter(poseStack, builder, sprite, color, combinedLight,
+                        combinedOverlay);
+            }
+        }
+
+        ItemStack moldStack = mold.getMoldStack();
+        IMold moldItem = IMold.get(moldStack);
+        if (moldItem != null) {
```

Comment:
I guess I'm wondering if it is efficient to compute this every render tick. It's fairly easy to cache values on block entities. Will wait for Alcatraz to give her opinion.